File: MemeApp.API/Models/Comment.cs
using System;
using System.Collections.Generic;

namespace MemeApp.API.Models
{
    public class Comment
    {
        public int Id { get; set; }
        public string Text { get; set; }

        public int Likes { get; set; }

        public int CommenterId { get; set; }

        public IList<CommentLike> LikeList { get; set; }

        public DateTime Created { get; set; }

        public Post Post { get; set; }

        public int PostId { get; set; }
    }
}

File: MemeApp-SPA/src/app/_models/Comment.ts
import { User } from './User';
import { Like } from './Like';
import { Post } from './Post';
import { CommentLike } from './CommentLike';
import { Reply } from './Reply';


export interface Comment {
    id: number;
    text: string;
    likes: number;
    commenterId: number;
    created: Date;
    postId: number;
    username: string;
    creatorId: number;
    photoUrl: string;
    likeList: CommentLike[];
    replies: Reply[];
    liked: boolean;
    deleteable: boolean;
    replying: boolean;
    replyText: string;
}
